pkg/devfile/parser/parse.go

```diff
@@ -478,6 +478,7 @@ func convertDevWorskapceTemplateToDevObj(dwTemplate v1.DevWorkspaceTemplate) (d
 
 //setDefaults sets the default values for nil boolean properties after the merging of devWorkspaceTemplateSpec is complete
 func setDefaults(d DevfileObj) (err error) {
+	schemaVersion := d.Data.GetSchemaVersion()
```

Comment:
`DevfileObj.Ctx.GetApiVersion()` should take higher priority than `Data.GetSchemaVersion()`. 

```
	if devfileVersion = DevfileObj.Ctx.GetApiVersion(); devfileVersion == "" {
		devfileVersion = DevfileObj.Data.GetSchemaVersion()
	}
```